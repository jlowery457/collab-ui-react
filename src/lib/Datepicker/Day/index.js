import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import * as utils from '@collab-ui/react/Datepicker/date_utils';

export default class Day extends React.Component {
  static displayName = 'Day';

  handleClick = event => {
    if (!this.isDisabled() && this.props.onClick) {
      this.props.onClick(event);
    }
  };

  handleKeyPress = e => {
    const { onClick } = this.props;
    e.preventDefault();
    if (e.which === 32 || e.which === 13) {
      return onClick(e);
    } else if (e.charCode === 32 || e.charCode === 13) {
      return onClick(e);
    }
  };

  handleMouseEnter = event => {
    if (!this.isDisabled() && this.props.onMouseEnter) {
      this.props.onMouseEnter(event);
    }
  };

  isSameDay = other => utils.isSameDay(this.props.day, other);

  isDisabled = () => utils.isDayDisabled(this.props.day, this.props);

  isOutsideMonth = () => {
    return (
      this.props.month !== undefined &&
      this.props.month !== utils.getMonth(this.props.day)
    );
  };

  isKeyboardSelected = () =>
    !this.isSameDay(this.props.selected) &&
    this.isSameDay(this.props.preSelection);

  getClassNames = () => {
    return classnames(
      'cui-datepicker__day',

      {
        'cui-datepicker__day--disabled': this.isDisabled(),
        'cui-datepicker__day--selected': this.isSameDay(this.props.selected),
        'cui-datepicker__day--keyboard-selected': this.isKeyboardSelected() && !this.isSameDay(utils.newDate()),
        'cui-datepicker__day--today': this.isSameDay(utils.newDate()) && !this.props.selected,
        'cui-datepicker__day--today--not-selected': this.props.selected && this.isSameDay(utils.newDate()) && !this.isSameDay(this.props.selected),
        'cui-datepicker__day--outside-month': this.isOutsideMonth()
      }
    );
  };

  render() {
    return (
      <div className='cui-datepicker__day-container'>
        <div
          className={this.getClassNames()}
          onClick={this.handleClick}
          onKeyPress={this.handleKeyPress}
          onMouseEnter={this.handleMouseEnter}
          aria-label={`day-${utils.getDate(this.props.day)}`}
          aria-selected={this.isSameDay(this.props.selected)}
          tabIndex='0' //TODO: (jlowery) not sure what to do with this.
          role='option'
        >
          {utils.getDate(this.props.day)}
        </div>
      </div>
    );
  }
}

Day.propTypes = {
  day: PropTypes.object.isRequired,
  month: PropTypes.number,
  onClick: PropTypes.func,
  onMouseEnter: PropTypes.func,
  preSelection: PropTypes.object,
  selected: PropTypes.object,
};