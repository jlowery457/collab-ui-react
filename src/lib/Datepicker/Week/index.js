import React from 'react';
import PropTypes from 'prop-types';
import Day from '@collab-ui/react/Datepicker/Day';
import * as utils from '@collab-ui/react/Datepicker/date_utils';

export default class Week extends React.Component {
  static displayName = 'Week';

  handleDayClick = (day, event) => {
    if (this.props.onDayClick) {
      this.props.onDayClick(day, event);
    }
  };

  handleDayMouseEnter = day => {
    if (this.props.onDayMouseEnter) {
      this.props.onDayMouseEnter(day);
    }
  };

  renderDays = () => {
    const startOfWeek = utils.getStartOfWeek(utils.cloneDate(this.props.day));
    const days = [];

    return days.concat(
      [0, 1, 2, 3, 4, 5, 6].map(offset => {
        const day = utils.addDays(utils.cloneDate(startOfWeek), offset);
        return (
          <Day
            key={offset}
            day={day}
            excludeDates={this.props.excludeDates}
            filterDate={this.props.filterDate}
            includeDates={this.props.includeDates}
            minDate={this.props.minDate}
            maxDate={this.props.maxDate}
            month={this.props.month}
            onClick={this.handleDayClick.bind(this, day)}
            onMouseEnter={this.handleDayMouseEnter.bind(this, day)}
            preSelection={this.props.preSelection}
            selected={this.props.selected}
          />
        );
      })
    );
  };

  render() {
    return <div className="cui-datepicker__week">{this.renderDays()}</div>;
  }
}

Week.propTypes = {
  day: PropTypes.object.isRequired,
  excludeDates: PropTypes.array,
  filterDate: PropTypes.func,
  includeDates: PropTypes.array,
  maxDate: PropTypes.object,
  minDate: PropTypes.object,
  month: PropTypes.number,
  onDayClick: PropTypes.func,
  onDayMouseEnter: PropTypes.func,
  preSelection: PropTypes.object,
  selected: PropTypes.object,
};