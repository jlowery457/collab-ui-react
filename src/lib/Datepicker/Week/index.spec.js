import React from 'react';
import { shallow } from 'enzyme';
import Week from '../Week';
import Day from '@collab-ui/react/Datepicker/Day';
import moment from 'moment';

const TEST_DATE = '2018-04-01';

describe('tests for <Week />', () => {
  const day = moment(TEST_DATE);
  it('should match week SnapShot', () => {
    const container = shallow(
      <Week day={day} month={3} />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render a div', () => {
    const container = shallow(
      <Week day={day} month={3} />
    );
    expect(container.find('div').hasClass('cui-datepicker__week')).toEqual(true);
  });

  it('should render 7 Day components', () => {
    const container = shallow(
      <Week day={day} month={3} />
    );

    expect(container.find(Day)).toHaveLength(7);
  });

  it('should handle day click event', () => {
    let count = 0;
    const container = shallow(
      <Week day={day} month={3} onDayClick={() => count++}/>
    );

    container.find(Day).first().simulate('click');
    expect(count).toEqual(1);
  });

  it('should handle day mouse enter event', () => {
    let count = 0;
    const container = shallow(
      <Week day={day} month={3} onDayMouseEnter={() => count++}/>
    );

    container.find(Day).first().simulate('mouseenter');
    expect(count).toEqual(1);
  });
});