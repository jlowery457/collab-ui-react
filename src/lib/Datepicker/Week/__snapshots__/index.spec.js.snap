// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <Week /> should match week SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Week
    day={"2018-04-01T05:00:00.000Z"}
    month={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Day
          day={"2018-04-01T05:00:00.000Z"}
          excludeDates={undefined}
          filterDate={undefined}
          includeDates={undefined}
          maxDate={undefined}
          minDate={undefined}
          month={3}
          onClick={[Function]}
          onMouseEnter={[Function]}
          preSelection={undefined}
          selected={undefined}
        />,
        <Day
          day={"2018-04-02T05:00:00.000Z"}
          excludeDates={undefined}
          filterDate={undefined}
          includeDates={undefined}
          maxDate={undefined}
          minDate={undefined}
          month={3}
          onClick={[Function]}
          onMouseEnter={[Function]}
          preSelection={undefined}
          selected={undefined}
        />,
        <Day
          day={"2018-04-03T05:00:00.000Z"}
          excludeDates={undefined}
          filterDate={undefined}
          includeDates={undefined}
          maxDate={undefined}
          minDate={undefined}
          month={3}
          onClick={[Function]}
          onMouseEnter={[Function]}
          preSelection={undefined}
          selected={undefined}
        />,
        <Day
          day={"2018-04-04T05:00:00.000Z"}
          excludeDates={undefined}
          filterDate={undefined}
          includeDates={undefined}
          maxDate={undefined}
          minDate={undefined}
          month={3}
          onClick={[Function]}
          onMouseEnter={[Function]}
          preSelection={undefined}
          selected={undefined}
        />,
        <Day
          day={"2018-04-05T05:00:00.000Z"}
          excludeDates={undefined}
          filterDate={undefined}
          includeDates={undefined}
          maxDate={undefined}
          minDate={undefined}
          month={3}
          onClick={[Function]}
          onMouseEnter={[Function]}
          preSelection={undefined}
          selected={undefined}
        />,
        <Day
          day={"2018-04-06T05:00:00.000Z"}
          excludeDates={undefined}
          filterDate={undefined}
          includeDates={undefined}
          maxDate={undefined}
          minDate={undefined}
          month={3}
          onClick={[Function]}
          onMouseEnter={[Function]}
          preSelection={undefined}
          selected={undefined}
        />,
        <Day
          day={"2018-04-07T05:00:00.000Z"}
          excludeDates={undefined}
          filterDate={undefined}
          includeDates={undefined}
          maxDate={undefined}
          minDate={undefined}
          month={3}
          onClick={[Function]}
          onMouseEnter={[Function]}
          preSelection={undefined}
          selected={undefined}
        />,
      ],
      "className": "cui-datepicker__week",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-01T05:00:00.000Z",
          "excludeDates": undefined,
          "filterDate": undefined,
          "includeDates": undefined,
          "maxDate": undefined,
          "minDate": undefined,
          "month": 3,
          "onClick": [Function],
          "onMouseEnter": [Function],
          "preSelection": undefined,
          "selected": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-02T05:00:00.000Z",
          "excludeDates": undefined,
          "filterDate": undefined,
          "includeDates": undefined,
          "maxDate": undefined,
          "minDate": undefined,
          "month": 3,
          "onClick": [Function],
          "onMouseEnter": [Function],
          "preSelection": undefined,
          "selected": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-03T05:00:00.000Z",
          "excludeDates": undefined,
          "filterDate": undefined,
          "includeDates": undefined,
          "maxDate": undefined,
          "minDate": undefined,
          "month": 3,
          "onClick": [Function],
          "onMouseEnter": [Function],
          "preSelection": undefined,
          "selected": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-04T05:00:00.000Z",
          "excludeDates": undefined,
          "filterDate": undefined,
          "includeDates": undefined,
          "maxDate": undefined,
          "minDate": undefined,
          "month": 3,
          "onClick": [Function],
          "onMouseEnter": [Function],
          "preSelection": undefined,
          "selected": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-05T05:00:00.000Z",
          "excludeDates": undefined,
          "filterDate": undefined,
          "includeDates": undefined,
          "maxDate": undefined,
          "minDate": undefined,
          "month": 3,
          "onClick": [Function],
          "onMouseEnter": [Function],
          "preSelection": undefined,
          "selected": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-06T05:00:00.000Z",
          "excludeDates": undefined,
          "filterDate": undefined,
          "includeDates": undefined,
          "maxDate": undefined,
          "minDate": undefined,
          "month": 3,
          "onClick": [Function],
          "onMouseEnter": [Function],
          "preSelection": undefined,
          "selected": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-07T05:00:00.000Z",
          "excludeDates": undefined,
          "filterDate": undefined,
          "includeDates": undefined,
          "maxDate": undefined,
          "minDate": undefined,
          "month": 3,
          "onClick": [Function],
          "onMouseEnter": [Function],
          "preSelection": undefined,
          "selected": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Day
            day={"2018-04-01T05:00:00.000Z"}
            excludeDates={undefined}
            filterDate={undefined}
            includeDates={undefined}
            maxDate={undefined}
            minDate={undefined}
            month={3}
            onClick={[Function]}
            onMouseEnter={[Function]}
            preSelection={undefined}
            selected={undefined}
          />,
          <Day
            day={"2018-04-02T05:00:00.000Z"}
            excludeDates={undefined}
            filterDate={undefined}
            includeDates={undefined}
            maxDate={undefined}
            minDate={undefined}
            month={3}
            onClick={[Function]}
            onMouseEnter={[Function]}
            preSelection={undefined}
            selected={undefined}
          />,
          <Day
            day={"2018-04-03T05:00:00.000Z"}
            excludeDates={undefined}
            filterDate={undefined}
            includeDates={undefined}
            maxDate={undefined}
            minDate={undefined}
            month={3}
            onClick={[Function]}
            onMouseEnter={[Function]}
            preSelection={undefined}
            selected={undefined}
          />,
          <Day
            day={"2018-04-04T05:00:00.000Z"}
            excludeDates={undefined}
            filterDate={undefined}
            includeDates={undefined}
            maxDate={undefined}
            minDate={undefined}
            month={3}
            onClick={[Function]}
            onMouseEnter={[Function]}
            preSelection={undefined}
            selected={undefined}
          />,
          <Day
            day={"2018-04-05T05:00:00.000Z"}
            excludeDates={undefined}
            filterDate={undefined}
            includeDates={undefined}
            maxDate={undefined}
            minDate={undefined}
            month={3}
            onClick={[Function]}
            onMouseEnter={[Function]}
            preSelection={undefined}
            selected={undefined}
          />,
          <Day
            day={"2018-04-06T05:00:00.000Z"}
            excludeDates={undefined}
            filterDate={undefined}
            includeDates={undefined}
            maxDate={undefined}
            minDate={undefined}
            month={3}
            onClick={[Function]}
            onMouseEnter={[Function]}
            preSelection={undefined}
            selected={undefined}
          />,
          <Day
            day={"2018-04-07T05:00:00.000Z"}
            excludeDates={undefined}
            filterDate={undefined}
            includeDates={undefined}
            maxDate={undefined}
            minDate={undefined}
            month={3}
            onClick={[Function]}
            onMouseEnter={[Function]}
            preSelection={undefined}
            selected={undefined}
          />,
        ],
        "className": "cui-datepicker__week",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-01T05:00:00.000Z",
            "excludeDates": undefined,
            "filterDate": undefined,
            "includeDates": undefined,
            "maxDate": undefined,
            "minDate": undefined,
            "month": 3,
            "onClick": [Function],
            "onMouseEnter": [Function],
            "preSelection": undefined,
            "selected": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-02T05:00:00.000Z",
            "excludeDates": undefined,
            "filterDate": undefined,
            "includeDates": undefined,
            "maxDate": undefined,
            "minDate": undefined,
            "month": 3,
            "onClick": [Function],
            "onMouseEnter": [Function],
            "preSelection": undefined,
            "selected": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-03T05:00:00.000Z",
            "excludeDates": undefined,
            "filterDate": undefined,
            "includeDates": undefined,
            "maxDate": undefined,
            "minDate": undefined,
            "month": 3,
            "onClick": [Function],
            "onMouseEnter": [Function],
            "preSelection": undefined,
            "selected": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-04T05:00:00.000Z",
            "excludeDates": undefined,
            "filterDate": undefined,
            "includeDates": undefined,
            "maxDate": undefined,
            "minDate": undefined,
            "month": 3,
            "onClick": [Function],
            "onMouseEnter": [Function],
            "preSelection": undefined,
            "selected": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-05T05:00:00.000Z",
            "excludeDates": undefined,
            "filterDate": undefined,
            "includeDates": undefined,
            "maxDate": undefined,
            "minDate": undefined,
            "month": 3,
            "onClick": [Function],
            "onMouseEnter": [Function],
            "preSelection": undefined,
            "selected": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-06T05:00:00.000Z",
            "excludeDates": undefined,
            "filterDate": undefined,
            "includeDates": undefined,
            "maxDate": undefined,
            "minDate": undefined,
            "month": 3,
            "onClick": [Function],
            "onMouseEnter": [Function],
            "preSelection": undefined,
            "selected": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-07T05:00:00.000Z",
            "excludeDates": undefined,
            "filterDate": undefined,
            "includeDates": undefined,
            "maxDate": undefined,
            "minDate": undefined,
            "month": 3,
            "onClick": [Function],
            "onMouseEnter": [Function],
            "preSelection": undefined,
            "selected": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
